src/rules/nitro/__tests__/correct-translation-usage.spec.js

```diff
@@ -0,0 +1,369 @@
+  valid: [
+    `
+  const MyComponent = () => (
+    <Button>
+      <Translate t="foo.submit" />
+    </Button>
+  );
+    `,
+    `
+  const MyComponent = () => (
+    <Button>
+      <Translate
+        t="foo.submit.placeholder_x"
+        values={{ x: 1337 }}
+      />
+    </Button>
+  );
+  `,
+    `
+  const MyComponent = () => (
+    <Button>
+      <Text t="foo.submit" />
+    </Button>
+  );
+    `,
+    `
+  const MyComponent = () => (
+    <Button>
+      <Text
+        t="foo.submit.placeholder_x"
+        values={{ x: 1337 }}
+      />
+    </Button>
+  );
+  `,
+    `
+    <Text
+      t="foo.two_placeholders"
+      values={{ days: 10, companyName: "Skypicker" }}
+    />
+  `,
+    `
+    <Text
+      t="foo.three_placeholders.two_same"
+      values={{ days: 10, companyName: "Skypicker" }}
+    />
+  `,
+    {
+      code: `<Translate t="new.key.without.final.copy" />`,
+      options: [
+        {
+          additionalTranslationsModule:
+            '@kiwicom/web-client/src/NEW_TRANSLATIONS.json'
+        }
+      ]
+    }
+  ],
+  invalid: [
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <Translate />
+    </Button>
+  );
+    `,
+      errors: ['A translation key needs to be specified.']
+    },
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <Text />
+    </Button>
+  );
+    `,
+      errors: ['A translation key needs to be specified.']
+    },
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <TextNode />
+    </Button>
+  );
+    `,
+      errors: ['A translation key needs to be specified.']
+    },
+    {
+      options: [
+        {
+          additionalTranslationsModule:
+            '@kiwicom/web-client/src/NEW_TRANSLATIONS.json'
+        }
+      ],
+      code: `<Translate t="unknown.key" />`,
+      errors: [
+        "'unknown.key' is not found in translations. Make sure the key was uploaded to PhraseApp and the package @kiwicom/translations is up to date (more details: https://kiwi.wiki/consumer-product/guides/translations/). If the copy is not final yet, the key can be added into @kiwicom/web-client/src/NEW_TRANSLATIONS.json."
+      ]
+    },
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <Translate t={foo} />
+    </Button>
+  );
+    `,
+      errors: ['The translation key needs to be a string literal.']
+    },
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <Translate t="foo" values={bar} />
+    </Button>
+  );
+    `,
+      errors: [
+        'The "values" attribute needs to be an object literal or object expression.'
+      ]
+    },
+    {
+      code: `
+  const MyComponent = () => (
+    <Button>
+      <Translate
+        t="foo.placeholder_x_foo"
+        values={{
+          x: 1337,
+          [foo]: "bar",
+        }}
+      />
+    </Button>
+  );
+    `,
+      errors: ['This key needs to be a string literal.']
```

Comment:
this key, what key? Would I know what I did wrong as a developer here?

Reply:
Yes. If you run it in the editor it should underline the specific problem. If you run it in the terminal it should also point you there